torchmetrics/functional/text/bleu.py

```diff
@@ -115,3 +101,48 @@ def bleu_score(
     bleu = brevity_penalty * geometric_mean
 
     return bleu
+
+
+def bleu_score(
+    reference_corpus: Sequence[Sequence[Sequence[str]]],
+    translate_corpus: Sequence[Sequence[str]],
+    n_gram: int = 4,
+    smooth: bool = False
+) -> Tensor:
+    """
+    Calculate BLEU score of machine translated text with one or more references
+
+    Args:
+        reference_corpus: An iterable of iterables of reference corpus
+        translate_corpus: An iterable of machine translated corpus
+        n_gram: Gram value ranged from 1 to 4
+        smooth: Whether or not to apply smoothing – Lin et al. 2004
+
+    Return:
+        Tensor with BLEU Score
+
+    Example:
+        >>> from torchmetrics.functional import bleu_score
+        >>> translate_corpus = ['the cat is on the mat'.split()]
+        >>> reference_corpus = [['there is a cat on the mat'.split(), 'a cat is on the mat'.split()]]
+        >>> bleu_score(translate_corpus, reference_corpus)
+        tensor(0.7598)
+
+    References:
+        [1] BLEU: a Method for Automatic Evaluation of Machine Translation by Papineni,
+        Kishore, Salim Roukos, Todd Ward, and Wei-Jing Zhu http://www.aclweb.org/anthology/P02-1040.pdf
+    """
+
+    if len(translate_corpus) != len(reference_corpus):
+        raise ValueError(f"Corpus has different size {len(translate_corpus)} != {len(reference_corpus)}")
+    numerator = torch.zeros(n_gram)
+    denominator = torch.zeros(n_gram)
+    c = tensor(0, dtype=torch.float)
+    r = tensor(0, dtype=torch.float)
+
+    c, r = _bleu_score_update(reference_corpus, translate_corpus, numerator, denominator, c, r, n_gram)
+
+    trans_len = c.clone().detach()
+    ref_len = r.clone().detach()
```

Comment:
why is the `clone().detach()` necessary?

Reply:
The initial implementation had this, I'm not really sure why we are doing it. It just makes a copy that is not considered in the computation graph of the model. We can use the same `c, r` I guess. Please let me know what I should do.

Reply:
Not doing the `clone().detach()` operation doesn't affect the tests.

Reply:
then I would remove them :]